rfcs/xxxx-v5-database.md

```diff
@@ -0,0 +1,169 @@
+
+<p align="center">
+	<img src="https://github.com/strapi/rfcs/assets/20578351/0d5841dd-89b1-42f6-bed5-b463f40f515b" width="600px" />
+</p>
+
+As an additional comment, the synched relation will only affect the locales inside the same publication state. Meaning, relations will not be shared between the `draft` and `published` versions of the document.
+
+<p align="center">
+	<img src="https://github.com/strapi/rfcs/assets/20578351/4a69c2e3-628e-45b1-8c77-338272a61c32" width="600px" />
+</p>
+
+
+### Polymorphic links
+
+V5 will eventually support polymorphic links.
+
+Those will allow use cases like Website Menus , linked to many other content types
+
+<p align="center">
+	<img src="https://github.com/strapi/rfcs/assets/20578351/29628824-c735-4776-b651-4c99ad11db8f" width="500px" />
+</p>
+
+This was just not possible in V4.
+
+## Link content schema
+
+Links will be represented as so in the content type schema:
+
+```jsx
+{
+  attributes: {
+    myLink: {
+      type: "link",
+      // Target one (or multiple) content types.
+      // Target all by not defining target
+      target: "uid-a" | ["uid-a", "uid-b"] | undefined,
+      multiple: true|false
+    }
+  }
+}
+```
+
+## Links Database Schema
+
+The database schema of `links` join tables :
+
+<p align="center">
+	<img src="https://github.com/strapi/rfcs/assets/20578351/f6ebb23e-c6e7-47b6-bdae-8030c80786a6" width="700px" />
+</p>
+
+The join table will now contain:
+
+- `Owner side doc id`: Reference to do the document id of the owner side (Article in the example)
+- `Inverse side doc id`: Reference ot the document id of the inverse side (Author in the example)
+- `pivot columns`: Reference the owner side variations (locale & publication state).
+- `order`: Only in `manyWays` and `morphMany` relations.
+
+## Query engine
+
+The query engine will expose the same methods as before.
+
+This layer will be used to interact with the database rows directly, and as so, methods such as `findOne` and `update` will reference entry ids and not document ids.
+
+Interactions with documents will be done in the application layer `DocumentService` , which will be documented in another RFC.
+
+# Tradeoffs
+
+Benefits of this approach:
+
+- Links schema reduces the number of queries needed to read and create relations.
+- It is conceptually easy, and similar to our approach.
+- It's easy to share the same relation between different document locales.
+
+Downsides:
+
+- Links will not allow for bidirectionality. A relation that references an entry to a document introduces many complexities that prevents bidirectionality, or makes it rather difficult. The first versions of links will not have bidirectionality in mind.
+
+# Unresolved questions
+
+- How do we manage life cycles?
+  - A database lifecycle would trigger at an entry level, not a document level. As everything will be document oriented, database life cycles might have less sense from a user perspective in V5.
```

Comment:
There should be a method to have lifecycles on document level as well.

Reply:
Indeed! We will have a document service (similar to the current entity service) with it's own lifecycles.